src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -461,10 +501,48 @@ fn check_exhaustive<'p, 'a: 'p, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
     }
 }
 
+fn maybe_point_at_variant(
+    cx: &mut MatchCheckCtxt<'a, 'tcx>,
+    sty: &TyKind<'tcx>,
+    patterns: &[Pattern<'_>],
+) -> Vec<Span> {
+    let mut covered = vec![];
+    if let ty::Adt(def, _) = sty {
+        // Don't point at the variants if they are too many to avoid visual clutter
```

Comment:
Confusing comment? If I understand correctly, the conditional on line 488 is guarding against too _many_ variant spans to point to, whereas this function is guarding against _overlapping_ spans (`.contains`, line 517)?

Reply:
changed to be clearer